hoot-core/src/main/cpp/hoot/core/io/OsmXmlReader.cpp

```diff
@@ -831,6 +831,28 @@ bool OsmXmlReader::startElement(const QString& /*namespaceURI*/, const QString&
             _element->setTag(key, value);
           }
         }
+        else if (key == MetadataTags::HootWayNodeCount())
+        {
+          bool ok;
+          size_t node_count = value.toULong(&ok);
+          if (!ok)
+          {
+            LOG_TRACE("Unable to parse '" << MetadataTags::HootWayNodeCount() << "' tag.");
+          }
+          if (_element->getElementType() == ElementType::Way)
+          {
+            WayPtr w = std::dynamic_pointer_cast<Way, Element>(_element);
+            if (w->getNodeCount() != node_count)
+            {
+              LOG_TRACE("Way node count tag does not equal actual way node count.");
+            }
+          }
+          else
+          {
+            LOG_TRACE(_element->getElementType().toString() << " elements shouldn't contain way node count tag.");
+          }
+          //  In the end, this tag isn't loaded because it is calculated
+        }
```

Comment:
The reader parses the tag and verifies it but doesn't use it.  No sense in propagating it into non-debug outputs.